conformance/lang/expressions/unary-expression/unary-minus-expression.balt

```diff
@@ -0,0 +1,984 @@
+
+    io:println(-a); // @output 1
+    io:println(-b); // @output 2
+    io:println((-c).toBalString()); // @output ()
+    io:println(-d); // @output 2147483648
+}
+
+Test-Case: output
+Description: Test unary negative expression for positive optional user defined sub types of int,
+              no overflow scenarios.
+Fail-Issue: ballerina-lang#31701
+Labels: DecimalNumber, module-type-defn, numeric-nil-lifting, optional-type, union-type, unary-minus, value:toBalString
+
+type Ints -2|-1|0|1|2;
+
+function init() {
+    Ints? a = 1;
+    Ints? b = ();
+    Ints? c = 0;
+
+    io:println(-a); // @output -1
+    io:println((-b).toBalString()); // @output ()
+    io:println(-c); // @output 0
+}
+
+Test-Case: output
+Description: Test unary negative expression for negative optional user defined sub types of int,
+              no overflow scenarios.
+Fail-Issue: ballerina-lang#31701
+Labels: DecimalNumber, module-type-defn, numeric-nil-lifting, optional-type, union-type, unary-minus, value:toBalString
+
+type Ints -2|-1|0|1|2;
+
+function init() {
+    Ints? a = -1;
+    Ints? b = ();
+    Ints? c = -2;
+
+    io:println(-a); // @output 1
+    io:println((-b).toBalString()); // @output ()
+    io:println(-c); // @output 2
+}
+
+Test-Case: output
+Description: Test unary negative expression for positive floats, no overflow scenarios.
+Label: float, DecimalNumber, unary-minus
+
+function init() {
+    float a = 1.0;
+    float b = 101.5;
+    float c = 0;
+    float d = 922439289.0;
+    io:println(-a); // @output -1.0
+    io:println(-b); // @output -101.5
+    io:println(-c); // @output -0.0
+    io:println(-d); // @output -9.22439289E8
+}
+
+Test-Case: output
+Description: Test unary negative expression for negative floats, no overflow scenarios.
+Label: float, DecimalNumber, unary-minus
+
+function init() {
+    float a = -1.0;
+    float b = -101.5;
+    float c = -922439289.0;
+    io:println(-a); // @output 1.0
+    io:println(-b); // @output 101.5
+    io:println(-c); // @output 9.22439289E8
+}
+
+Test-Case: output
+Description: Test unary negative expression for very large floating point number.
+Label: float, DecimalNumber, unary-minus
+
+function init() {
+    float a = 25.E1742f;
+    float b = -a;
+    io:println(b); // @output -Infinity
+}
+
+Test-Case: output
+Description: Test unary negative expression for positive user defined sub types of float,
+              no overflow scenarios.
+Fail-Issue: ballerina-lang#32928
+Labels: DecimalNumber, module-type-defn, unary-minus, union-type
+
+type Floats -2.0|-1.0|0|1.0|2.0;
+
+function init() {
+    Floats a = 1.0;
+    Floats b = 2.0;
+    Floats c = 0;
+
+    io:println(-a); // @output -1.0
+    io:println(-b); // @output -2.0
+    io:println(-c); // @output -0.0
+}
+
+Test-Case: output
+Description: Test unary negative expression for user defined sub types of float
+Fail-Issue: ballerina-lang#32928
+Labels: DecimalNumber, module-type-defn, module-const-decl, unary-minus, union-type
```

Comment:
Let's add `FloatingPointTypeSuffix` label wherever which uses `FloatTypeSuffix ` or `DecimalTypeSuffix`

Reply:
Updated in 79de921